src/rust/inetstack/protocols/tcp/queue.rs

```diff
@@ -435,81 +427,12 @@ impl<const N: usize> SharedTcpQueue<N> {
         }
     }
 
-    pub fn receive(
-        &mut self,
-        ip_hdr: Ipv4Header,
-        tcp_hdr: TcpHeader,
-        local: SocketAddrV4,
-        remote: SocketAddrV4,
-        buf: DemiBuffer,
-    ) {
+    pub fn receive(&mut self, ip_hdr: Ipv4Header, tcp_hdr: TcpHeader, buf: DemiBuffer) {
         // If this queue has an allocated receive queue, then direct the packet there.
         if let Some(recv_queue) = self.recv_queue.as_mut() {
             recv_queue.push((ip_hdr, tcp_hdr, buf));
             return;
         }
-
```

Comment:
You still need to generate a reset here because we don't have a socket data structure to do further processing. I will eventually just move PassiveOpenSocket to this file as well (or at least squash it all down to one receive queue).